lib/shore/client/messaging/v1/base.rb

```diff
@@ -9,3 +9,16 @@ class Base < JsonApiClient::Resource # :nodoc:
     end
   end
 end
+
+class ShoreClientMiddleware < Faraday::Middleware # :nodoc:
+  def call(env)
+    env[:request_headers]['Authorization'] = Shore::Client.access_token
+    fail Shore::Client::Tokens::InvalidTokenError,
```

Comment:
You don't need to do this. The server will give you a 401. In fact, you shouldn't do this. We might have some public apis that don't require any authentication. In fact, change the above line to:
```
if (token = Shore::Client.access_token)
   env[:request_headers]['Authorization'] = "Bearer #{token}"
end
@app.call(env)
```